subsys/net/ip/connection.c

```diff
@@ -47,6 +47,37 @@ LOG_MODULE_REGISTER(net_conn, CONFIG_NET_CONN_LOG_LEVEL);
 
 static struct net_conn conns[CONFIG_NET_MAX_CONN];
 
+static sys_slist_t conn_unused;
+static sys_slist_t conn_used;
+
+static struct net_conn *conn_get_unused(void)
+{
+	sys_snode_t *node;
+
+	node = sys_slist_peek_head(&conn_unused);
+	if (!node) {
+		return NULL;
+	}
+
+	sys_slist_remove(&conn_unused, NULL, node);
+
+	return CONTAINER_OF(node, struct net_conn, node);
+}
+
+static void conn_set_used(struct net_conn *conn)
+{
+	conn->flags |= NET_CONN_IN_USE;
+
+	sys_slist_prepend(&conn_used, &conn->node);
+}
+
+static void conn_set_unused(struct net_conn *conn)
+{
+	(void)memset(conn, 0, sizeof(*conn));
+
+	sys_slist_prepend(&conn_unused, &conn->node);
```

Comment:
Why not call here `sys_slist_remove(&conn_used, NULL, &conn->node);` too?

Reply:
It could. It means conn_get_unused() would not remove the conn from the unused list itself. Sure why not